### PR TITLE
replacement of caddy-docker

### PR DESCRIPTION
**Pull request by lihongan**
replacing the old caddy server with new nginx server.

1. the nginx-alpine image size is smaller than caddy-docker ( 9M vs 43M)
2. one nginx-alpine image replaces other four caddy-docker images (caddy-docker, caddy-docker-2, caddy-docker-3 and caddy-docker-4), so no need to mirror many images for disconnected cluster.
3. nginx has more options to support SAN and SNI than caddy.
4. nginx probably can replace other images like caddy-docker-wildcard/httbin/header-test in feature (under researching)

since some files in testdata are still used by other teams, will delete them when all replacement is done.

@quarterpin please help review. thanks

### CODE REVIEW

**Comment by lihongan** [comment]
logs:
https://mastern-jenkins-csb-openshift-qe.cloud.paas.psi.redhat.com/job/Runner-v3-smoke/2590/console
https://mastern-jenkins-csb-openshift-qe.cloud.paas.psi.redhat.com/job/Runner-v3-smoke/2589/console
https://mastern-jenkins-csb-openshift-qe.cloud.paas.psi.redhat.com/job/Runner-v3-smoke/2588/console
https://mastern-jenkins-csb-openshift-qe.cloud.paas.psi.redhat.com/job/Runner-v3-smoke/2587/console
https://mastern-jenkins-csb-openshift-qe.cloud.paas.psi.redhat.com/job/Runner-v3-smoke/2585/console  (two failed cases are unrelated to this PR (know issue) and tracked by https://issues.redhat.com/browse/OCPQE-2816 )

File: features/routing/abrouting.feature [hunk 1]
```diff
@@ -11,28 +11,21 @@ Feature: Testing abrouting
...
-    Given I obtain test data file "routing/abrouting/unseucre/service_unsecure.json"
-    When I run the :create client command with:
-      | f | service_unsecure.json |
-    Then the step should succeed
-    Given I obtain test data file "routing/abrouting/unseucre/service_unsecure-2.json"
-    When I run the :create client command with:
-      | f | service_unsecure-2.json |
-    Then the step should succeed
-    Given I wait for the "service-unsecure" service to become ready
-    Given I wait for the "service-unsecure-2" service to become ready
```
Comment (lihongan) [inline features/routing/abrouting.feature:35]: the new abtest-websrv[12].yaml already created the services, above steps can be removed.

File: features/routing/idle.feature [hunk 2]
```diff
@@ -4,28 +4,28 @@ Feature: idle service related scenarios
   # @case_id OCP-10935
   Scenario: Pod can be changed to un-idle when there is unsecure or edge or passthrough route coming
     Given I have a project
-    Given I obtain test data file "routing/list_for_caddy.json"
-    When I run oc create over "list_for_caddy.json" replacing paths:
-      | ["items"][0]["spec"]["replicas"] | 1 |
+    Given I obtain test data file "routing/web-server-rc.yaml"
+    When I run the :create client command with:
+      | f | web-server-rc.yaml |
```
Comment (lihongan) [inline features/routing/idle.feature:9]: the new web-server-rc.yaml uses `replicas: 1` as default, replacing paths is not required.

File: features/routing/haproxy-router.feature [hunk 3]
```diff
@@ -174,12 +174,12 @@ Feature: Testing haproxy router
 
     Given I switch to the first user
     And I have a project
-    Given I obtain test data file "routing/list_for_caddy.json"
-    When I run the :create client command with:
-      | f | list_for_caddy.json |
+    Given I obtain test data file "routing/web-server-rc.yaml"
+    When I run oc create over "web-server-rc.yaml" replacing paths:
+      | ["items"][0]["spec"]["replicas"] | 2 |
```
Comment (lihongan) [inline features/routing/haproxy-router.feature:179]: the old list_of_caddy.json uses `replicas: 2` and the new web-server-rc.yaml uses `replicas: 1`, but the `backend health check interval` cases need at least two backends, so `replacing paths` is required here.

File: features/routing/abrouting.feature [hunk 4]
```diff
@@ -99,40 +92,33 @@ Feature: Testing abrouting
...
+      | name       | route-reencrypt                           |
+      | hostname   | <%= rand_str(5, :dns) %>-reen.example.com |
+      | service    | service-secure                            |
+      | cert       | example_wildcard.pem                      |
+      | key        | example_wildcard.key                      |
+      | cacert     | route_reencrypt.ca                        |
+      | destcacert | route_reencrypt_dest.ca                   |
     Then the step should succeed
     When I run the :annotate client command with:
       | resource     | route                                          |
```
Comment (quarterpin) [inline features/routing/abrouting.feature:124]: Spacing for `|` can be corrected and modified.

File: features/routing/abrouting.feature [hunk 5]
```diff
@@ -192,13 +178,13 @@ Feature: Testing abrouting
...
+      | f | abtest-websrv1.yaml |
     Then the step should succeed
-    Given I obtain test data file "routing/abrouting/unseucre/service_unsecure-2.json"
+    Given I obtain test data file "routing/abrouting/abtest-websrv2.yaml"
     When I run the :create client command with:
-      | f | service_unsecure-2.json |
+      | f | abtest-websrv2.yaml |
     Then the step should succeed
     When I run the :create_route_edge client command with:
       | name    | edge1            |
```
Comment (quarterpin) [inline features/routing/abrouting.feature:190]: Spacing for `|` can be corrected and modified. (same as above)

File: features/routing/abrouting.feature [hunk 6]
```diff
@@ -321,17 +307,17 @@ Feature: Testing abrouting
     # Scale pods
     When I run the :scale client command with:
       | resource | replicationcontrollers |
-      | name     | test-rc-1              |
+      | name     | abtest-websrv1         |
```
Comment (quarterpin) [inline features/routing/abrouting.feature:310]: Spacing for `|` can be corrected and modified. (same as above)

File: features/routing/abrouting.feature [hunk 7]
```diff
@@ -321,17 +307,17 @@ Feature: Testing abrouting
...
       | replicas | 2                      |
     Then the step should succeed
     When I run the :scale client command with:
       | resource | replicationcontrollers |
-      | name     | test-rc-2              |
+      | name     | abtest-websrv2         |
       | replicas | 4                      |
```
Comment (quarterpin) [inline features/routing/abrouting.feature:316]: Spacing for `|` can be corrected and modified. (same as above)

File: features/routing/abrouting.feature [hunk 8]
```diff
@@ -321,17 +307,17 @@ Feature: Testing abrouting
...
     Then the step should succeed
     When I run the :scale client command with:
       | resource | replicationcontrollers |
-      | name     | test-rc-3              |
+      | name     | abtest-websrv3         |
       | replicas | 3                      |
```
Comment (quarterpin) [inline features/routing/abrouting.feature:321]: Spacing for `|` can be corrected and modified.

File: features/routing/abrouting.feature [hunk 9]
```diff
@@ -403,17 +389,17 @@ Feature: Testing abrouting
...
       | resource | replicationcontrollers |
-      | name     | test-rc-1              |
+      | name     | abtest-websrv1         |
       | replicas | 2                      |
     Then the step should succeed
     When I run the :scale client command with:
       | resource | replicationcontrollers |
-      | name     | test-rc-2              |
+      | name     | abtest-websrv2         |
       | replicas | 4                      |
```
Comment (quarterpin) [inline features/routing/abrouting.feature:398]: same as above for `|`

File: features/routing/abrouting.feature [hunk 10]
```diff
@@ -403,17 +389,17 @@ Feature: Testing abrouting
...
     Then the step should succeed
     When I run the :scale client command with:
       | resource | replicationcontrollers |
-      | name     | test-rc-3              |
+      | name     | abtest-websrv3         |
       | replicas | 3                      |
```
Comment (quarterpin) [inline features/routing/abrouting.feature:403]: same as above for `|`

**Comment by quarterpin** [comment]
/lgtm 

Thanks, @lihongan .

**Comment by lihongan** [comment]
@liangxia @pruan-rht please help take a look. thanks.

**Review by pruan-rht — COMMENTED** [review]
/lgtm

**Pull request merged**